src/main/java/de/fraunhofer/iais/eis/ids/connector/handlers/standard/DescriptionRequestHandler.java

```diff
@@ -31,74 +32,59 @@
 
     public DescriptionRequestHandler(DynamicConnectorSelfDescription selfDescription, DapsSecurityTokenProvider daps, LoggingInteractor loggingInteractor) {
         this.selfDescription = selfDescription;
         this.daps = daps;
         this.loggingInteractor = loggingInteractor;
     }
 
-    
+
     @Override
     public DescriptionResponseMAP handle(DescriptionRequestMAP DescriptionRequestMAP) throws RejectMessageException {
         String result;
         DescriptionRequestMAP.getMessage().getRequestedElement();
-        List<String> resourceList = new ArrayList<String>();
         loggingInteractor.logMaP((MessageAndPayload) DescriptionRequestMAP);
         //currently it is wrapped inside
         BaseConnector infrastructureComponent = (BaseConnector) selfDescription.getSelfDescription();
         ResourceCatalog resourceCatalog = infrastructureComponent.getResourceCatalog().get(0); //Attention. There must always be a catalog, or this will fail
-        resourceIdInfrastructureComponent = resourceCatalog.getOfferedResource().get(0).getId(); //Attention. What if catalog is empty? Exception would be thrown here
-        String sampleURIKey = null;
-        result = infrastructureComponent.toRdf();
-        if (DescriptionRequestMAP.getMessage().getRequestedElement() != null) {
-
-            sampleURIKey = DescriptionRequestMAP.getMessage().getRequestedElement().toString();
-            
-            
-            
-            //review 1
-            //If asked for artifact 
-            //Object sampleObject = (ResourceCatalog)infrastructureComponent.getResourceCatalog().get(0).getOfferedResource().get(0);
-       
-            List<Resource> resourceIdList = (List<Resource>) infrastructureComponent.getResourceCatalog().get(0).getOfferedResource();
-            Iterator<Resource> itr = resourceIdList.iterator();
-            while (itr.hasNext() && itr != null) {
-                Resource resourceId = (Resource) itr.next();
-                resourceList.add(resourceId.getId().toString());
-            }
-            try {
-            	String tempVal = resourceCatalog.getId().toString();        
-                if (!tempVal.isEmpty() && resourceList.size()>0) {
-                    //fetch response only through resource catalogID
-                	//if asked for resource id
-                	
-                	for (int i = 0; i < resourceList.size(); i++) { 
-                		
-                		//fetch response through resourceId ,but unable to fetch single element!!!!!!!!!!!
-                        //TODO: value ignored
-                		result = infrastructureComponent.getResourceCatalog().get(0).getOfferedResource().get(i).toRdf();
-                	}
-                	
-                    //query not sure if i have to change the response
+        String URIKey = null;
+        result = "empty";
+        try{
+            URIKey = DescriptionRequestMAP.getMessage().getRequestedElement().toString();
```

Comment:
Beware NullPointer here! The requested element can be null, which is then interpreted as a "self-description request"

Reply:
I just noticed you catch NullPointer below. Well, I guess that works - it just doesn't look that nice :D